JSON output for get and list commands

fkorotkov opened the pull request:
In the light of the upcoming `1.0.0` release and stabilizing of the API, let's introduce some breaking changes for the good.

Removed all the `--cpu`, `--memory`, `--disk` and `--display` flags and replaced with a single `--json` flag for machine-readable output.

Added `--json` option to the `list` command to output a single JSON list. Notably removed `--quite` flag since it seemed unnecessary.

Fixes #297

fkorotkov commented:
FYI @tomjn @fmoc @hokadiri @josephburnett

edigaryev commented:
1. I do think it's beneficial to keep the `-q` and `--quiet` for simple use-cases like https://github.com/cirruslabs/tart/discussions/375#discussioncomment-4611062.

    I would never recommend anyone to parse table-like output that the `tart list` provides as this will likely break in the future (https://github.com/cirruslabs/tart/issues/297#issuecomment-1385814735), and processing `--json` output with `jq` might be too complicated for this every-day task.

2. IMO the VM size should be printed in human-understandable units and follow the source and name of the VM. `tart list` output when running this PR:

    ```
    Source Size Name                                                                                                         
    local  19   macos                                                                                                        
    oci    50   ghcr.io/cirruslabs/macos-ventura-base:latest
    ```

    Proposed `tart list` output:

    ````
    Source Name                                               Size    
    local  macos                                              18.45 GB
    oci    ghcr.io/cirruslabs/macos-ventura-base:latest       46.6 GB
    ````

4. The `--table/--json Output format (default: table)` option looks a bit weird, perhaps simply adding a `--json` flag should be enough?

5. There seems to be an extraneous newline emitted at the end when using `tart get` and `tart list`.

tomjn commented:
> and processing --json output with jq might be too complicated for this every-day task.

Without `jq` parsing it in shell script/bash becomes **significantly** harder, `jq` makes it much easier. If you're going to add JSON don't piecemeal it just add it everywhere in a consistent manner.

Use of JSON should also be an indicator that the intended recipient is inhuman

> IMO the VM size should be printed in human-understandable units and follow the source and name of the VM. 

This sounds like a great excuse for a flag/parameter and a new issue+PR

> The --table/--json Output format (default: table) option looks a bit weird, perhaps simply adding a --json flag should be enough?

Lets not forget YAML or real ASCII tables ( not pretend TSV's ), why not a `format` parameter?

WP CLI uses an optional `--format` parameter that works on all commands

```sh
wp post list # the default, same as table
wp post list --format=table
wp post list --format=json
wp post list --format=yaml
```

It also allows for `-q` and `--quiet`.

This is possible because there's a separation between doing things and generating output, with dedicated utilities just for formatting raw data into presentable text

https://make.wordpress.org/cli/handbook/references/internal-api/wp-cli-utils-format-items/

fmoc commented:
Docker for instance provides pretty-printed JSON when using `inspect`, which works well enough in the rare occasions you need to read those attributes manually. (With `jq`, you can even color-format the output, but in my opinion some default formatting/pretty-printing should be included so one does not depend on this tool.)

fkorotkov commented:
Thank you all for the feedback. I've addressed all of it except human readable disk size for the `text` output format. IMO for JSON is should be just an Integer value but I haven't found how to make formatting customization for `text` vs `json` output.

@edigaryev I've also reverted `--quite` flag in [f1dbe3e](https://github.com/cirruslabs/tart/pull/394/commits/f1dbe3ead60e485cc8b91fbe14afdd6d9677d444). Fixed new line in [b836429](https://github.com/cirruslabs/tart/pull/394/commits/b836429d5a8a1aa3df462613217913fed885f8b7). PTAL.

edigaryev commented on Sources/tart/Commands/Get.swift:
```diff
@@ -1,52 +1,31 @@
-      throw ValidationError("Options --cpu, --memory, --disk-size and --display are mutually exclusive")
-    }
-  }
+  @Option(help: "Output format")
```
```suggestion
@Option(help: "Output format: text or json")
```

edigaryev commented on Sources/tart/Commands/List.swift:
```diff
@@ -2,15 +2,24 @@ import ArgumentParser
   @Option(help: ArgumentHelp("Only display VMs from the specified source (e.g. --source local, --source oci)."))
   var source: String?
 
+  @Option(help: "Output format")
```
```suggestion
  @Option(help: "Output format: text or json")
```
